docs/reference/files/charmcraft-yaml-file.rst

```diff
@@ -567,6 +615,12 @@ for users) and the default is 0 (root).
         :start-at: links:
         :end-before: name:
 
+.. admonition:: Best Practice
+    :class: hint
+
+    Documentation links should apply to the charm, and not to the application that
+    is being charmed. Assume that the user already has basic competency in the use
+    of the application.
```

Comment:
Note for the future: We'll want to add one more thing to this note, that is, links to examples of nicely documented charms.